# DP: XXX missing description
# DP: 
# DP: Comment from Chet Ramey <chet@nike.ins.cwru.edu>:
# DP: 
# DP: Nope.  This will allow setuid scripts if not called as `sh' and not 
# DP: called with the -p option.  I won't install this. 
 

--- a/bash/shell.c
+++ b/bash/shell.c
@@ -486,7 +486,7 @@
   if (dump_translatable_strings)
     read_but_dont_execute = 1;
 
-  if (running_setuid && privileged_mode == 0)
+  if (running_setuid && privileged_mode == 0 && act_like_sh == 0)
     disable_priv_mode ();
 
   /* Need to get the argument to a -c option processed in the
